lib/spack/env/cc

```diff
@@ -301,6 +306,18 @@ case "$mode" in
         args=("${args[@]}" ${SPACK_LDLIBS[@]}) ;;
 esac
 
+# Prepend SPACK_DYNAMIC_LINKER & add it's directory to rpath
+if [[ -n "$SPACK_DYNAMIC_LINKER" && -e "$SPACK_DYNAMIC_LINKER" ]]; then
+    dynamic_linker_dir="$(dirname $SPACK_DYNAMIC_LINKER)"
+    if [[ $mode == ccld ]]; then
+        $add_rpaths && args=("$dynamic_linker$SPACK_DYNAMIC_LINKER" "${args[@]}")
+        $add_rpaths && args=("-Wl,-rpath=$dynamic_linker_dir" "${args[@]}")
```

Comment:
Note how the `-Wl,-rpath=` syntax is slightly different from the normal `-Wl,-rpath,`, so I couldn't use `$rpath`. I am wondering if we should export yet another variable like `$SPACK_DYNAMIC_LINKER_RPATH` or something similar.